### PR TITLE
 add periodic sync trigger

### PR DESCRIPTION
**Pull request by pidah**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```feature user

```

### CODE REVIEW

**Comment by gardener-robot-ci-3** [comment]
Thank you @pidah for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

File: controllers/authentication/openidconnect_controller.go [hunk 1]
```diff
@@ -35,6 +35,8 @@ type OpenIDConnectReconciler struct {
 	*unionAuthTokenHandler
 }
 
+var ResyncPeriod time.Duration
```
Comment (mvladev) [inline controllers/authentication/openidconnect_controller.go:38]: Don't expose this as global variable - add this to the controller struct and have `AddFlags` and `ApplyTo` functions that can be used to add the flags and apply the value of those flags to the configuration

**Review by mvladev — CHANGES_REQUESTED** [review]

**Comment by mvladev** [comment]
Rebase this PR on top of the `master` branch not the upcoming offline issuer branch

**Review by mvladev — COMMENTED** [review]
With this change, you need to remove the predicate or this won't work as expected

```
WithEventFilter(predicate.GenerationChangedPredicate{}).
```

https://github.com/gardener/oidc-webhook-authenticator/pull/9/files#diff-abfd17e4d40d2b8ddcbad0d9d43f682557f066f9daf7a0761bba03807b357072L169

**Comment by pidah** [comment]
I can remove, but from my testing, the periodic sync triggers reconciliation when the spec is unchanged and the metadata field is changed.

**Review by mvladev — APPROVED** [review]

**Pull request merged**